### PR TITLE
Fix NPE on projectile source within ModifyBowProjectile module, Resolves #57

### LINKED ISSUE #57: Enderpearl bow projectile error
Read: https://github.com/Electroid/PGM/issues/57#issuecomment-551205212

~~Add custom projectiles so maps like Ozone works properly, and so we can have cool snowball bows on Glacial Impact;~~

```xml
<modifybowprojectile>
    <projectile>EnderPearl</projectile>
    <velocityMod>2.5</velocityMod>
</modifybowprojectile>
```
```xml
<modifybowprojectile>
    <projectile>Snowball</projectile>
    <velocityMod>2.0</velocityMod>
</modifybowprojectile>
```

### PR DESCRIPTION
**Pull request by botinator**
Signed-off-by: botinator <53882853+botinator@users.noreply.github.com>

This is related to PR #106, this is the iteration based on review feedback. Used the web editor for that version so decided to just split it into another PR

### CODE REVIEW

**Comment by botinator** [comment]
This also fixes an unreported bug where custom projectiles weren't reporting the appropriate ownership on death. See Discord for more information.

File: src/main/java/tc/oc/pgm/modules/ModifyBowProjectileMatchModule.java [hunk 1]
```diff
@@ -25,13 +21,13 @@
 
 @ListenerScope(MatchScope.RUNNING)
 public class ModifyBowProjectileMatchModule extends MatchModule implements Listener {
-  protected final Class<? extends Entity> cls;
+  protected final Class<? extends Projectile> cls;
```
Comment (Pablete1234) [inline src/main/java/tc/oc/pgm/modules/ModifyBowProjectileMatchModule.java]: PGM allows for entities to be shot, not necessarily just projectiles, I don't think that functionality should be removed.

Comment (botinator) [reply src/main/java/tc/oc/pgm/modules/ModifyBowProjectileMatchModule.java]: If this is something we're interested in keeping I can address it in the next couple of days, my thought process here was to simplify the solution using a native method that we know works. A lot of these custom entities have problems, for example: deaths caused by explosions, splash potion events, etc. I left the projectile definitions as is (entities) so this limit is just for this specific module.

Comment (Electroid) [reply src/main/java/tc/oc/pgm/modules/ModifyBowProjectileMatchModule.java]: Although entities were technically supported, it was usually used to do weird stuff like creeper bows -- which caused other sorts of bugs.

To fix the ender pearl issue, I'm okay merging this fix.

**Pull request merged**